dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordPropertyMapping.cs

```diff
@@ -0,0 +1,179 @@
+            Type t when t == typeof(ulong) || t == typeof(ulong?) => (ulong)reader.GetInt64(propertyIndex),
+            Type t when t == typeof(short) || t == typeof(short?) => reader.GetInt16(propertyIndex),
+            Type t when t == typeof(ushort) || t == typeof(ushort?) => (ushort)reader.GetInt16(propertyIndex),
+            Type t when t == typeof(bool) || t == typeof(bool?) => reader.GetBoolean(propertyIndex),
+            Type t when t == typeof(float) || t == typeof(float?) => reader.GetFloat(propertyIndex),
+            Type t when t == typeof(double) || t == typeof(double?) => reader.GetDouble(propertyIndex),
+            Type t when t == typeof(decimal) || t == typeof(decimal?) => reader.GetDecimal(propertyIndex),
+            Type t when t == typeof(DateTime) || t == typeof(DateTime?) => reader.GetDateTime(propertyIndex),
+            Type t when t == typeof(string) => reader.GetString(propertyIndex),
+            Type t when t == typeof(byte[]) => (byte[])reader[propertyName],
+            Type t when t == typeof(ReadOnlyMemory<float>) || t == typeof(ReadOnlyMemory<float>?) => GetReadOnlyMemory((byte[])reader[propertyName]),
+            _ => throw new NotSupportedException($"Unsupported type: {propertyType} for property: {propertyName}")
+        };
+    }
+
+    #region private
+
+    private static ReadOnlyMemory<float> GetReadOnlyMemory(byte[] byteArray)
+    {
+        var array = MemoryMarshal.Cast<byte, float>(byteArray).ToArray();
+        return new ReadOnlyMemory<float>(array);
+    }
+
+    private static string GetStorageDataPropertyType(VectorStoreRecordProperty property)
+    {
+        return property.PropertyType switch
+        {
+            // Integer types
+            Type t when t == typeof(int) || t == typeof(int?) => "INTEGER",
+            Type t when t == typeof(long) || t == typeof(long?) => "INTEGER",
+            Type t when t == typeof(ulong) || t == typeof(ulong?) => "INTEGER",
+            Type t when t == typeof(short) || t == typeof(short?) => "INTEGER",
+            Type t when t == typeof(ushort) || t == typeof(ushort?) => "INTEGER",
+
+            // Floating-point types
+            Type t when t == typeof(float) || t == typeof(float?) => "REAL",
+            Type t when t == typeof(double) || t == typeof(double?) => "REAL",
+            Type t when t == typeof(decimal) || t == typeof(decimal?) => "REAL",
+
+            // String type
+            Type t when t == typeof(string) => "TEXT",
+
+            // Boolean types - SQLite doesn't have a boolean type, represent it as 0/1
+            Type t when t == typeof(bool) || t == typeof(bool?) => "INTEGER",
+
+            // Byte array (BLOB)
+            Type t when t == typeof(byte[]) => "BLOB",
+
+            // DateTime types - use ISO 8601 string for datetimes
+            Type t when t == typeof(DateTime) || t == typeof(DateTime?) => "TEXT",
```

Comment:
I'm unsure about making assumptions like this.  E.g. someone may specify datetime in their data model and point it to an INTEGER column with a unix time in it and expect it to work.  For other dbs I stuck strictly to the types they support without extra conversion.  That said, my data type validation is also only done in the mapper, so if someone wanted to they could build a mapper with more assumptions and more data types.